### PR TITLE
Support for table prefix

### LINKED ISSUE #1014: Installation with existing database?
How does one go about using this package with an existing app/database. Seems to be only good for a brand new database.

?

### PR DESCRIPTION
**Pull request by MrCrayon**
Use --with-prefix=prefix_ to install all voyager tables with a prefix. (Need to add documentation)

Running the whole test suite with prefix gives no error but I can't guaranty that everything works so I would tag the functionality as experimental.

- Change all voyager tables with the use of helper get_prefix_table
- Left unchanged slugs because that will affect also templates
- Prefixable Trait applies prefix to table name
- Added a test to verify install with prefix works

Closes #1014

### CODE REVIEW

File: migrations/2016_01_01_000000_add_voyager_user_fields.php [hunk 1]
```diff
@@ -9,8 +9,8 @@ class AddVoyagerUserFields extends Migration
      */
     public function up()
     {
-        Schema::table('users', function ($table) {
-            if (!Schema::hasColumn('users', 'avatar')) {
+        Schema::table(get_prefixed_table('users'), function ($table) {
```
Comment (fletch3555) [inline migrations/2016_01_01_000000_add_voyager_user_fields.php:12]: We **_really_** shouldn't be changing old migrations... Works well for new users, but could very easily burn existing users trying to upgrade.

Comment (MrCrayon) [reply migrations/2016_01_01_000000_add_voyager_user_fields.php:12]: I understand your concern but I don't think we would be able to add prefix support without changing old migrations.

If that can help whenever you are updating or rolling back `get_prefix_table` helper will output the same table name passed if voyager.database.table_prefix is not set (as it should for previous installations) or empty (default value for new installations).

So, unless I'm missing something, it shouldn't change anything for old users unless they willingly add a table_prefix.

**Comment by sergiogregorutti** [comment]
This is a nice feature. What's the status of this? Can we use it on new projects?

**Comment by emptynick** [comment]
Closing as this will be implemented in Voyager 2

**Comment by minutmorph** [comment]
Hi @emptynick , is there an expected release date for V2? thanks

**Comment by cyberfab007** [comment]
Why would such a feature sit on a shelf for almost 2 years ?

**Pull request closed without merging**